File: apps/backend-go/internal/model/page.go
package model

import (
	"encoding/json"
	"time"
)

// Page 对应页面实体（含 schema JSON）。
//
// Schema 使用 json.RawMessage 而非 []byte：[]byte 经 encoding/json 序列化会被
// base64 编码成字符串，破坏 BFF/website 对 schema 的嵌套对象解析（Java 端是
// JsonNode 嵌套对象）。json.RawMessage 是 []byte 的别名，但实现了 MarshalJSON
// 直接原样嵌入 JSON，从而输出嵌套对象。
//
// V2-T2: Seo 持久化页面级 SEO（与 schema 同样的 RawMessage 处理）。
type Page struct {
	ID        string          `db:"id" json:"id"`
	SiteID    string          `db:"site_id" json:"siteId"`
	Name      string          `db:"name" json:"name"`
	Path      string          `db:"path" json:"path"`
	Status    string          `db:"status" json:"status"`
	Schema    json.RawMessage `db:"schema_json" json:"schema"`
	Seo       json.RawMessage `db:"seo_json" json:"seo"`
	CreatedAt time.Time       `db:"created_at" json:"createdAt"`
	UpdatedAt time.Time       `db:"updated_at" json:"updatedAt"`
}

File: apps/backend-java/src/main/java/com/luban/backend/entity/Page.java
package com.luban.backend.entity;

import java.time.Instant;

/**
 * Page entity; table pages. schemaJson holds JSON string.
 *
 * V2-T2: seoJson 持久化页面级 SEO（PageSchema.seo 结构）。
 */
public class Page {
    private String id;
    private String siteId;
    private String name;
    private String path;
    private String status;
    private String schemaJson;
    /** V2-T2 页面级 SEO JSON title/description/keywords/og/canonical/noIndex */
    private String seoJson;
    private Instant createdAt;
    private Instant updatedAt;

    public String getId() { return id; }
    public void setId(String id) { this.id = id; }
    public String getSiteId() { return siteId; }
    public void setSiteId(String siteId) { this.siteId = siteId; }
    public String getName() { return name; }
    public void setName(String name) { this.name = name; }
    public String getPath() { return path; }
    public void setPath(String path) { this.path = path; }
    public String getStatus() { return status; }
    public void setStatus(String status) { this.status = status; }
    public String getSchemaJson() { return schemaJson; }
    public void setSchemaJson(String schemaJson) { this.schemaJson = schemaJson; }
    public String getSeoJson() { return seoJson; }
    public void setSeoJson(String seoJson) { this.seoJson = seoJson; }
    public Instant getCreatedAt() { return createdAt; }
    public void setCreatedAt(Instant createdAt) { this.createdAt = createdAt; }
    public Instant getUpdatedAt() { return updatedAt; }
    public void setUpdatedAt(Instant updatedAt) { this.updatedAt = updatedAt; }
}
